File: Capítulo_4/Pilhas/Criando uma classe JavaScript Stack baseada Obj/Pop de elementos da pilha/pop-pilha.js
// teremos de implementar manualmente a lógica para remover um elemento. O método pop também devolve o elemento que foi removido da pilha. Esse método é implementado assim:
class Stack {
    constructor() {
        this.count = 0;
        this.items = {};
    }
    pop() {
        if (this.isEmpty()) {
            return undefined;
        }
        this.count--;
        const result = this.items[this.count];
        delete this.items[this.count];
        return result;
    }
}

File: Capítulo_4/Pilhas/Criando uma classe JavaScript Stack baseada Obj/Pop de elementos da pilha/Código em Python/pop-pilha.py
class Stack:
    def __init__(self):
        self.count = 0
        self.items = {}

    def pop(self):
        if (self.count <= 0):
            return None
        self.count -= 1
        result = self.items[self.count]
        del self.items[self.count]
        return result
